binder/views.py

```diff
@@ -1158,10 +1169,51 @@ def _filter_relation(self, field_name, where_map, request, include_annotations):
 		return FilterDescription(q, need_distinct)
 
 
+	# handle self.alternative_filters as part of _parse_filter
+	def _parse_alternative_filters(self, field, filter_heads, *args, **kwargs):
+		head, tail = re.fullmatch(r'([^.:]*)(.*)', field).groups()
+
+		# split any/all of the tail, by default a filter is treated as any
+		#   NOTE: not is_any ==> is_all
+		is_any = bool(re.search(':any', tail))
+		is_all = bool(re.search(':all', tail))
+		is_not = bool(re.search(':not', tail))
+		if is_any and is_all:
+			raise BinderRequestError('Cannot use both :all and :any in alternative filter "{}".'.format(field))
+		# clean :all and :any from tail
+		tail = tail.replace(':all', '').replace(':any', '')
```

Comment:
Technically a filter can go through multiple alternative filters, thus `:all` and `:any` can appear at multiple places in the tail. Now you don't necessarily only parse the any/all of the current alternative filter.

I think its best to only match `:all` and `:any` if the tail starts with it. So for example:
```python
is_any = tail.startswith(':any')
if is_any:
    tail = tail[4:]
```

etc